llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -333,7 +362,19 @@ buildFunctionsToAspectsMap(Module &M, TypeToAspectsMapTy &TypesWithAspects) {
 PreservedAnalyses
 SYCLPropagateAspectsUsagePass::run(Module &M, ModuleAnalysisManager &MAM) {
   TypeToAspectsMapTy TypesWithAspects = getTypesThatUseAspectsFromMetadata(M);
-  propagateAspectsToOtherTypesInModule(M, TypesWithAspects);
+  AspectValueToNameMapTy AspectValues = getSYCLAspectsFromMetadata(M);
+
+  // If there is no metadata for aspect values the source code must not have
+  // included the SYCL headers. In that case there should also not be any types
+  // that use aspects, so we can skip this pass.
+  if (AspectValues.empty()) {
+    assert(TypesWithAspects.empty() &&
+           "sycl_aspects metadata is missing but "
+           "intel_types_that_use_aspects is present.");
```

Comment:
I suppose this needs renaming at some point to sycl_?

Reply:
Good eye! Yes, I might as well do that renaming now.